Added work experience projects what they say about me sections

DevTaehong opened the pull request:
- Added Work experience, Featured Projects, and What they say about me sections
- Minor changes in other components

![Screenshot 2023-09-11 at 6 10 43 PM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/0bb6ce16-3878-4e36-a930-8f1b3cfcb064)
![Screenshot 2023-09-11 at 6 11 00 PM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/567a11b9-ded6-4a74-a900-ce4e1a606704)
![Screenshot 2023-09-11 at 6 11 09 PM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/7143596c-cce6-4f90-b36f-a3516ba80b5e)

cdsalty commented on components/featuredProjects/FeaturedProjects.tsx:
Great job!

cdsalty commented on components/featuredProjects/ProjectCard.tsx:
I don't believe you need "use clent" here... You only want to bring this in when it's necessary. I am including a link to the docs that has a great comparison of when to use a Server Component and when to use a Client Component.
[When to use Server and Client Components?](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#when-to-use-server-and-client-components)

Instead of hardcoding the tech stack items, you can use Array.map to dynamically generate them.
Use Array.map for the Tech Stack section. 
```
<div className="flex flex-row gap-1 truncate xl:gap-5">
        {techStack.map((tech, i) => (
          <div key={i} className="smallRegular techStack flex items-start text-white900">
            {tech}
          </div>
        ))}
      </div>
```

DevTaehong replied:
![Screenshot 2023-09-13 at 12 26 19 AM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/9bef88ef-057c-4d93-a140-55a8b48de0a6)

This is an error I got when I removed the 'use client'. I believe that I got that because I'm using NextUI's Card component. Is there another way to deal with it?

I refactored the techStack times!

cdsalty commented on components/featuredProjects/ProjectCard.tsx:
```diff
@@ -0,0 +1,96 @@
+
+import { arrow } from "@/public";
+
+type projectProps = {
```
The type definition projectProps should follow the TypeScript conventions of PascalCase for type and interface names. The correct convention is `ProjectProps`

cdsalty commented on components/featuredProjects/ProjectCard.tsx:
```diff
@@ -0,0 +1,96 @@
+}) => {
+  return (
+    <Card
+      key={project.name}
```
A 'key' prop is not necessary here. It's not part of an array map function. You can remove this.

DevTaehong replied:
Removed it.

cdsalty commented on components/myServices/ServiceCard.tsx:
```diff
@@ -0,0 +1,64 @@
+  return (
+    <>
+      {serviceIcons.map((ServiceIcon, index) => (
+        <motion.div
```
Same thing here...

DevTaehong replied:
![Screenshot 2023-09-13 at 1 11 51 AM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/85318349-292c-484f-b148-5b96368b169b)

I'm having this missing key prop error here. Did you mean this??

DevTaehong replied:
I've made a reusable component called Service and mapped over it. Thanks!

cdsalty commented on components/myServices/ServiceIProvide.tsx:
```diff
@@ -0,0 +1,21 @@
+const ServiceIProvide = () => {
+  return (
+    <section className="flex flex-col items-center gap-10 bg-white900 py-12 dark:bg-black300 sm:bg-white800 xl:px-[5.31rem] xl:pb-[8.9375rem] xl:pt-[4.5rem]">
+      <h1 className="mySkills relative dark:text-white900 xl:text-[3rem] xl:tracking-[-0.03rem]">
```
Why are you using a className called "mySkills" inside "ServiceIProvide?" Maybe you can give it a new name, that is more in aligned with it being used in more than one place?

DevTaehong replied:
My mistake! Thanks.

cdsalty commented on components/workExperience/CompanyCard.tsx:
```diff
@@ -0,0 +1,50 @@
+  return (
+    <>
+      {companies.map((CompanyLogo, index) => (
+        <motion.div
```
The same thing here. map over and return a reusable component here. Also, there is no reason to use a key as you currently have it. If this is necessary for NextUi, please forgive my suggestion...

DevTaehong replied:
I've made a reusable component and map over that. Thanks!

cdsalty commented on components/workExperience/SquareWorkExperienceCard.tsx:
```diff
@@ -0,0 +1,31 @@
+"use client";
```
Not necessary here..

DevTaehong replied:
![Screenshot 2023-09-13 at 2 07 23 AM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/8a60b160-152b-4a43-835f-ff5f3f23bb1c)
When I removed that, I got this error. I think it's related to NextUI components

cdsalty commented on components/workExperience/WorkExperience.tsx:
Apologies as I am unable to run this locally currently, but be sure the responsiveness is working correctly. Be sure to check all layouts... Thanks!

DevTaehong replied:
I'm sure the responsiveness is working properly. Thank you for your feedback!

cdsalty reviewed (changes requested):
It's coming along nicely @DevTaehong! Take a look at the changes I requested. Please let me know if you need any clarification or assistance with anything. Thanks!!

cdsalty commented:
@Nassim-Saboundji, will you please review the Merging is blocked error?

DevTaehong commented:
> @Nassim-Saboundji, will you please review the Merging is blocked error?

Sorry, It was me. I added a protect rule for the branch. But, I removed it. Thanks!

DevTaehong commented:
@cdsalty I've made the requested changes. Thank you for your feedback!

cdsalty reviewed (approved):
It's looking great @DevTaehong! Thank you for taking the time to consider the suggestions I shared. I want to request that you create a separate, dedicated branch specifically aimed at addressing the light-mode issue, particularly in the "Feature Projects" section where white text is currently used. 
Besides that, it looks great, and I can't wait to see it come along! I'm sorry again for the delay. You are more than welcome to reach out to me anytime there is a delay or any other concern you have. We're all here for you!
